lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/SnappyAirCompressorCodec.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.avro;
+
+import io.airlift.compress.v3.snappy.SnappyCompressor;
+import io.airlift.compress.v3.snappy.SnappyDecompressor;
+import org.apache.avro.file.Codec;
+import org.apache.avro.file.CodecFactory;
+import org.apache.avro.file.DataFileConstants;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import java.util.zip.CRC32;
+
+public class SnappyAirCompressorCodec
+        extends Codec
+{
+    public static final CodecFactory SNAPPY_CODEC_FACTORY = new CodecFactory()
+    {
+        @Override
+        protected Codec createInstance()
+        {
+            return new SnappyAirCompressorCodec();
+        }
+    };
+
+    private final SnappyCompressor compressor = SnappyCompressor.create();
```

Comment:
nit: this could be just `private final Compressor` and `private final Decompressor`

Reply:
Need to be class specific on Decompressor to get access to `io.airlift.compress.v3.snappy.SnappyDecompressor#getUncompressedLength` so I keep compressor specific too to match. Can change